frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices/install-lso-sc.tsx

```diff
@@ -0,0 +1,152 @@
+import * as React from 'react';
+import { match } from 'react-router';
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore
+import { useDispatch } from 'react-redux';
+import { ActionGroup, Button, Form } from '@patternfly/react-core';
+import {
+  NodeKind,
+  StorageClassResourceKind,
+  referenceForModel,
+  k8sCreate,
+  K8sResourceKind,
+} from '@console/internal/module/k8s';
+import {
+  withHandlePromise,
+  HandlePromiseProps,
+  history,
+  ButtonBar,
+} from '@console/internal/components/utils';
+import { setFlag } from '@console/internal/actions/features';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+import { getName } from '@console/shared';
+import { NO_PROVISIONER, minSelectedNode, defaultRequestSize } from '../../../constants';
+import { OCSServiceModel } from '../../../models';
+import AttachedDevicesNodeTable from './sc-node-list';
+import { PVsAvailableCapacity } from '../pvs-available-capacity';
+import { OCS_CONVERGED_FLAG, OCS_FLAG } from '../../../features';
+import { makeLabelNodesRequest } from '../create-form';
+import { lvsResource } from '../../../constants/resources';
+import { getOCSRequestData } from '../ocs-request-data';
+import {
+  OCSAlert,
+  SelectNodesSection,
+  StorageClassSection,
+} from '../../../utils/common-ocs-install-el';
+
+import '../ocs-install.scss';
+import './attached-devices.scss';
+
+const makeOCSRequest = (
+  selectedData: NodeKind[],
+  storageClass: StorageClassResourceKind,
+): Promise<any> => {
+  const promises = makeLabelNodesRequest(selectedData);
+  const scName = getName(storageClass);
+  const ocsObj = getOCSRequestData(scName, defaultRequestSize.BAREMETAL);
+
+  return Promise.all(promises).then(() => k8sCreate(OCSServiceModel, ocsObj));
+};
+
+export const CreateOCS = withHandlePromise<CreateOCSProps & HandlePromiseProps>((props) => {
+  const {
+    handlePromise,
+    errorMessage,
+    inProgress,
+    match: {
+      params: { appName, ns },
+    },
+  } = props;
+  const [filteredNodes, setFilteredNodes] = React.useState<string[]>([]);
+  const [storageClass, setStorageClass] = React.useState<StorageClassResourceKind>(null);
+  const [nodes, setNodes] = React.useState<NodeKind[]>([]);
+  const [lvsList, setLvsList] = React.useState<K8sResourceKind[]>([]);
+  const dispatch = useDispatch();
+  const [lvsData, lvsLoaded, lvsLoadError] = useK8sWatchResource<K8sResourceKind[]>(lvsResource);
+
+  const handleStorageClass = (sc: StorageClassResourceKind) => {
+    setStorageClass(sc);
+
+    if (sc) {
+      const [lvs] = lvsList.filter((l) => l?.spec?.storageClassName === sc?.metadata?.name);
+      setFilteredNodes(
+        lvs?.spec?.nodeSelector?.nodeSelectorTerms?.[0]?.matchExpressions?.[0]?.values,
```

Comment:
As per the design `0` element only give the nodes associated with a SC.

Reply:
The label selector can take multiple forms: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#nodeselector-v1-core

Up to you if you want to handle only the form that you create. Is this view only for LocalVolumeSets created from the UI?

@gnehapk I thought you weren't using the hostname label?

Reply:
The node labels are required to identify the nodes associated with a SC. Else how UI can figure that out? Do you have any suggestion/idea? @rohantmp

Reply:
As discussed with @rohantmp  this will handle all the local volume set that will be created from UI. If LVS is created from CLI, there can be many complex combinations w.r.t label which needs further discussion. So will handle the CLI scenario in a follow up PR if required.